Ginger/GingerCoreCommon/Repository/NewRepositorySerializer.cs

```diff
@@ -587,18 +584,38 @@ private static void SetRepositoryItemHeaderAttr(RepositoryItemHeader RIH, string
         private static void xmlReadListOfObjects(object ParentObj, XmlReader xdr, IObservableList observableList)
         {
             // read list of object into the list, add one by one, like activities, actions etc.
-
             if (ParentObj is RepositoryItemBase && ((RepositoryItemBase)ParentObj).ItemBeenReloaded)
             {
                 observableList.Clear();//clearing existing list items in case it is been reloaded
             }
-
             //TODO: Think/check if we want to make all observe as lazy load
-            if (LazyLoadAttr.Contains(xdr.Name) && observableList.AvoidLazyLoad == false)
+            LazyLoadListConfig lazyLoadConfig = LazyLoadListConfigs.Where(x => x.ListName == xdr.Name).FirstOrDefault();
+            if (ParentObj is RepositoryItemBase && lazyLoadConfig != null && observableList.AvoidLazyLoad == false)
             {
-                // We can save line/col and reload later when needed
-                string s = xdr.ReadOuterXml(); // .ReadInnerXml(); // .Read();
-                observableList.DoLazyLoadItem(s);
+                observableList.LazyLoadDetails = new LazyLoadListDetails();
+                observableList.LazyLoadDetails.Config = new LazyLoadListConfig() { ListName = lazyLoadConfig.ListName, LazyLoadType = lazyLoadConfig.LazyLoadType };//copying the values so original config won't get changed
+                switch(observableList.LazyLoadDetails.Config.LazyLoadType)
+                {
+                    case LazyLoadListConfig.eLazyLoadType.NodePath:
+                        if (string.IsNullOrEmpty(((RepositoryItemBase)ParentObj).FilePath) == false 
+                                && File.Exists(((RepositoryItemBase)ParentObj).FilePath)
+                                    && ((RepositoryItemBase)ParentObj).DirtyStatus != Common.Enums.eDirtyStatus.Modified)
+                        {
+                            observableList.LazyLoadDetails.XmlFilePath = ((RepositoryItemBase)ParentObj).FilePath;
+                            xdr.ReadOuterXml();//so xdr will progress
+                        }
+                        else //can't go with NodePath approch because no file to refernce or file do not have latest data
+                        {
+                            observableList.LazyLoadDetails.Config.LazyLoadType = LazyLoadListConfig.eLazyLoadType.StringData;
+                            observableList.LazyLoadDetails.DataAsString = xdr.ReadOuterXml();
+                        }
+                        break;
+
+                    case LazyLoadListConfig.eLazyLoadType.StringData:
```

Comment:
Codacy found an issue: [Remove this empty 'case' clause.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=5848147)